javascript/packages/cli/src/actions/convert.ts

```diff
@@ -1,117 +1,159 @@
+function convertConfig(config: PolkadotLaunch.LaunchConfig): ZombienetConfig {
+  const relaychain = convertRelaychain(config.relaychain);
+  const parachains = convertParachains(
+    config.simpleParachains,
+    config.parachains,
+  );
+  const hrmpChannels = convertHrmpChannels(config.hrmpChannels);
 
-  const {
+  return {
     relaychain,
-    parachains = [],
-    simpleParachains = [],
-    hrmpChannels = [],
-    types,
-  } = convertedJson;
+    parachains,
+    hrmp_channels: hrmpChannels,
+    types: config.types,
+  };
+}
 
-  const nodes: NodeConfig[] = [];
-  let paras: ParachainConfig[] = [];
+function convertRelaychain(
+  relaychain: PolkadotLaunch.LaunchConfig["relaychain"],
+): ZombienetConfig["relaychain"] {
+  const { chain, genesis, nodes = [], bin } = relaychain;
 
-  const DEFAULT_NODE_VALUES = {
+  const convertedNodes = nodes.map((node) => ({
+    name: node.name,
+    args: node.flags,
+    ws_port: node.wsPort,
+    rpc_port: node.rpcPort,
+    p2p_port: node.port,
+    balance: DEFAULT_BALANCE,
     validator: true,
     invulnerable: true,
-    balance: DEFAULT_BALANCE,
+  }));
+
+  return {
+    chain,
+    default_command: bin,
+    genesis,
+    nodes: convertedNodes,
   };
+}
 
-  paras = paras.concat(
-    parachains.map(({ id, nodes }) => ({
-      id,
-      collators: ((nodes as PL_NodesConfig[]) || []).map(({ name }) => ({
-        name,
-        command: "adder-collator",
-        ...DEFAULT_NODE_VALUES,
-      })),
-    })),
+function convertParachains(
+  simpleParachains: PolkadotLaunch.LaunchConfig["simpleParachains"] = [],
+  parachains: PolkadotLaunch.LaunchConfig["parachains"] = [],
+): ZombienetConfig["parachains"] {
+  const convertedSimpleParachains = simpleParachains.map(
+    convertSimpleParachain,
   );
+  const convertedParachains = parachains.map(convertParachain);
 
-  paras = paras.concat(
-    simpleParachains.map(({ id, name }) => ({
-      id,
-      collators: [{ name, command: "adder-collator", ...DEFAULT_NODE_VALUES }],
-    })),
-  );
+  return convertedSimpleParachains.concat(convertedParachains);
+}
 
-  if (relaychain?.nodes) {
-    relaychain.nodes.forEach((n: any) => {
-      nodes.push({
-        name: `"${n.name}"`,
-        ...DEFAULT_NODE_VALUES,
-      });
-    });
-  }
+function convertSimpleParachain(
+  simpleParachain: ArrayElement<
+    PolkadotLaunch.LaunchConfig["simpleParachains"]
+  >,
+): ArrayElement<ZombienetConfig["parachains"]> {
+  const { id, balance, port, bin } = simpleParachain;
 
-  const jsonOutput: PolkadotLaunchConfig = {
-    relaychain: {
-      default_image: "docker.io/paritypr/polkadot-debug:master",
-      default_command: "polkadot",
-      default_args: ["-lparachain=debug"],
-      chain: relaychain?.chain || "",
-      nodes,
-      genesis: relaychain?.genesis,
-    },
-    types,
-    hrmp_channels: hrmpChannels,
-    parachains: paras,
+  const collator = {
+    name: "alice",
+    command: bin,
+    p2p_port: +port,
```

Comment:
```suggestion
    p2p_port: +port,
```
nice usage of unary operator!